src/WCSimDetectorConfigs.cc

```diff
@@ -463,41 +463,201 @@ void WCSimDetectorConstruction::SetNuPrismBeamTest_mPMTGeometry()
 	vessel_cyl_height = 38.*CLHEP::mm;    //option A, option B would be 277 mm
 	vessel_radius_curv = 342.*CLHEP::mm;  //needs to include the vessel thickness, as we construct from outside inwards.
 	vessel_radius = 254.*CLHEP::mm;
-	dist_pmt_vessel = 8*CLHEP::mm;      
+	dist_pmt_vessel = 2*CLHEP::mm;      
 	orientation = PERPENDICULAR;
 	mPMT_outer_material = "G4_PLEXIGLASS";
 	mPMT_inner_material = "Air";               // TODO: real air, hence update abs_length
 	mPMT_material_pmtAssembly = "SilGel";
 	mPMT_outer_material_d = 10*CLHEP::mm;
 	
 	// Radius of cone at z=reflectorHeight
-	id_reflector_height = 9.29*CLHEP::mm;        // From solidworks CAD drawings
-	id_reflector_z_offset = 6.12*CLHEP::mm;
-	id_reflector_angle = 46.5*CLHEP::deg; 
-	mPMT_pmt_openingAngle = 8.3*CLHEP::deg;
+	id_reflector_height = 6.53*CLHEP::mm;        // for a radius of 7.25mm, for hex: 5.4mm (radius of 6mm)
+	id_reflector_z_offset = 4.8*CLHEP::mm;       //from KM3Net CAD drawings
+	id_reflector_angle = 48.*CLHEP::deg;         // Need to be remeasured for different PMT curvature 
+	mPMT_pmt_openingAngle = 8.7*CLHEP::deg;     // for hex: 8.5deg
 	G4double vessel_tot_height = vessel_radius + vessel_cyl_height;
 	
 	// parameters related to filling the ID mPMT
 	nID_PMTs = 19;
 	config_file = wcsimdir_path+"/mPMT-configfiles/mPMTconfig_19_nuPrism_3ring.txt"; // for smaller reflector, use: mPMTconfig_19_nuPrism.txt (hex)
 
-	WCIDHeight               = 3.38*CLHEP::m;
-    	WCIDDiameter             = 3.696*CLHEP::m;
-    	WCIDVerticalPosition     = 0.;
+	WCIDHeight               = 3.38*CLHEP::m; //original
+	WCIDDiameter             = 3.696*CLHEP::m; //  original
+	WCIDVerticalPosition     = 0.;
 	
 	WCBarrelPMTOffset     = 300.*CLHEP::mm;
     WCPMTperCellHorizontal = 1.0; // 1 per phi
     WCPMTperCellVertical   = 1.0;
 
     WCBarrelNumPMTHorizontal = 18;
     WCBarrelNRings        = 5;
-    //WCCapPMTSpacing       = (pi*WCIDDiameter/WCBarrelNumPMTHorizontal);
+	//WCCapPMTSpacing       = (pi*WCIDDiameter/WCBarrelNumPMTHorizontal);
     WCCapPMTSpacing       = 0.58*CLHEP::m;
+	//	WCCapPMTSpacing       = 0.7*CLHEP::m; //test
     WCCapEdgeLimit        = 1.551*m; 
     WCBlackSheetThickness = 2.0*cm;    // deprecate soon.
     WCAddGd               = false;
 }
 
+//original WCTE but dimensions from  CAD, L.Anthony
```

Comment:
Lots of duplicated code in the following functions too. Could you create a single function instead that takes an argument for switching?

Reply:
Will do

Reply:
i think i meant the mPMT parameters, as in my latest review summary, but this is broader issue that should be solved in a future PR